src/Language/Avro/Types.hs

```diff
@@ -25,9 +26,9 @@ instance Semigroup Protocol where
     Protocol
       (ns p1)
       (pname p1)
-      (imports p1 <> imports p2)
-      (types p1 <> types p2)
-      (messages p1 <> messages p2)
+      (nub (imports p1 <> imports p2))
```

Comment:
I am thinking that maybe we can keep [sets](http://hackage.haskell.org/package/containers-0.6.2.1/docs/Data-Set.html) or even [hash sets](http://hackage.haskell.org/package/unordered-containers-0.2.10.0/docs/Data-HashSet.html) instead of lists which are then de-duplicated.

Reply:
Good point! I'll merge this as is right now and tomorrow I'll play with the Set idea, thanks! 😄